src/components/Menu/index.tsx

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-self-assign */
+
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Text, Button, Input, SelectRate } from 'components';
+import { CgClose } from 'react-icons/cg';
+import { AiOutlineSave } from 'react-icons/ai';
+import DatePicker from 'components/Pickers/DatePicker';
+import TimePicker from 'components/Pickers/TimePicker';
+import DurationPickers from 'components/Pickers/DurationPicker';
+
+import { Program, ProgramRating } from 'services/epg';
+import {
+  BottomContainer,
+  ButtonContainer,
+  ContentContainer,
+  Form,
+  FormContainer,
+  MenuContainer,
+  MenuStyleProps,
+  Toolbar,
+} from './styles';
+
+export interface MenuProps extends MenuStyleProps {
+  minWidth;
+  overflowStatus;
+  programs: Program[];
+  selectedProgramId: string;
+  setisClosing: (programId: boolean) => void;
+}
+
+const Menu: React.FC<MenuProps> = ({
+  minWidth,
+  overflowStatus,
+  programs,
+  selectedProgramId,
+  setisClosing,
+}) => {
+  const { t } = useTranslation();
+
+  const [program, setProgram] = React.useState<Program[]>(programs);
+
+  let i = 0;
+  let aux;
+  let title;
+  let editTitle;
+  let description;
+  let rating;
+  let defaultRate;
+  let date;
+  let time;
+  let timeLength;
+
+  while (i < programs.length) {
+    if (programs[i].id === selectedProgramId) {
+      aux = i;
+      title = programs[i].title;
+      editTitle = title.toString();
+      description = programs[i].description;
+      rating = programs[i].rating;
+      defaultRate = rating.toString();
+      date = new Date(programs[i].startDate);
+      time = new Date(programs[i].startHour);
+      timeLength = programs[i].duration;
+    }
+    // eslint-disable-next-line no-plusplus
+    ++i;
+  }
```

Comment:
Install `@ungap/structured-clone` and `@types/ungap__structured-clone`.

```suggestion
  const program = programs.find(p => p.id === selectedProgramId);

  const [newProgram, setNewProgram] = React.useState<Program | undefined>(
    structuredClone(program),
  );
```